### PR TITLE
refactor windowed part 3

### PR DESCRIPTION
**Pull request by SergBobrovsky**
smaller, faster, successfully pass tests )
flake8 is evil (

### CODE REVIEW

File: more_itertools/more.py [hunk 1]
```diff
@@ -672,8 +672,8 @@ def unique_to_each(*iterables):
     return [list(filter(uniques.__contains__, it)) for it in pool]
 
 
-def windowed(seq, n, fillvalue=None, step=1):
-    """Return a sliding window of width *n* over the given iterable.
+def windowed(iterable, width, fillvalue=None, step=1):
```
Comment (bbayles) [inline more_itertools/more.py]: I'd prefer if the parameter names didn't change.

File: more_itertools/more.py [hunk 2]
```diff
@@ -694,43 +694,28 @@ def windowed(seq, n, fillvalue=None, step=1):
...
+                yield window
+        if 0 < len(window) < width:
+            yield chain(window, (fillvalue,) * (width - len(window)))
+        elif 0 < i < min(step, width):
+            window.extend((fillvalue,) * i)
+            yield window
+
+    if width >= 0 < step:
+        return map(tuple, helper(**locals()) if width else ((),))
+    raise ValueError('width must be >= 0\nstep must be > 0')
```
Comment (bbayles) [inline more_itertools/more.py]: This loses the specific error messages of the original implementation.

File: more_itertools/more.py [hunk 3]
```diff
@@ -694,43 +694,28 @@ def windowed(seq, n, fillvalue=None, step=1):
...
+                i = step
+                yield window
+        if 0 < len(window) < width:
+            yield chain(window, (fillvalue,) * (width - len(window)))
+        elif 0 < i < min(step, width):
+            window.extend((fillvalue,) * i)
+            yield window
+
+    if width >= 0 < step:
+        return map(tuple, helper(**locals()) if width else ((),))
```
Comment (bbayles) [inline more_itertools/more.py]: Is it really necessary to inline `locals` here? What's the performance impact?

File: more_itertools/more.py (hunk 2 above)
Comment (SergBobrovsky) [reply more_itertools/more.py]: I do not mind that you take only the helper() and implement it in the previous version of the windowed, keeping in it everything that is dear to you.

File: more_itertools/more.py (hunk 1 above)
Comment (SergBobrovsky) [reply more_itertools/more.py]: The discussion of **n** versus **width** seems like a waste of time for me, if you believe that somewhere there is a dude who passes a named parameter **n** to the windowed() - your right to save this argument name.

File: more_itertools/more.py (hunk 3 above)
Comment (SergBobrovsky) [reply more_itertools/more.py]: This is my jamb, I was tired there ) Only **step** is critical, so changing the signature of the helper to a **helper(step)** will be right.

**Comment by SergBobrovsky** [comment]
I don't mind to you improve pr as you wish.

**Pull request merged**